ui/app/services/path-help.js

```diff
@@ -168,42 +211,45 @@ export default Service.extend({
     });
   },
 
-  getNewAdapter(paths, itemType) {
+  getNewAdapter(pathInfo, itemType) {
     //we need list and create paths to set the correct urls for actions
-    const { list, create, apiPath } = paths;
-    const createPath = create.find(path => path.path.includes(itemType));
-    const listPath = list.find(pathInfo => pathInfo.path.includes(itemType));
-    const deletePath = paths.delete.find(path => path.path.includes(itemType));
+    let paths = this.filterPathsByItemType(pathInfo, itemType);
+    let { apiPath } = pathInfo;
+    const getPath = paths.find(path => path.operations.includes('get'));
+
+    //the action might be "Generate" or something like that so we'll grab the first post endpoint if there
+    //isn't one with "Create"
+    //TODO: look into a more sophisticated way to determine the create endpoint
+    const createPath = paths.find(path => path.action === 'Create' || path.operations.includes('post'));
+    const deletePath = paths.find(path => path.operations.includes('delete'));
+
     return generatedItemAdapter.extend({
       urlForItem(method, id) {
-        let { path } = listPath;
-        let url = `${this.buildURL()}/${apiPath}${path.slice(1)}/`;
+        let url = `${this.buildURL()}/${apiPath}${getPath.path.slice(1)}/`;
         if (id) {
           url = url + encodePath(id);
         }
         return url;
       },
 
-      urlForFindRecord(id, modelName, snapshot) {
-        return this.urlForItem(modelName, id, snapshot);
+      urlForFindRecord(id, modelName) {
+        return this.urlForItem(modelName, id);
       },
 
+      //urlForQuery if there is an id and we are listing, use the id to construct the path
       urlForUpdateRecord(id) {
-        let { path } = createPath;
-        path = path.slice(1, path.indexOf('{') - 1);
+        let path = createPath.path.slice(1, createPath.path.indexOf('{') - 1);
         return `${this.buildURL()}/${apiPath}${path}/${id}`;
       },
 
       urlForCreateRecord(modelType, snapshot) {
         const { id } = snapshot;
-        let { path } = createPath;
-        path = path.slice(1, path.indexOf('{') - 1);
+        let path = createPath.path.slice(1, createPath.path.indexOf('{') - 1);
         return `${this.buildURL()}/${apiPath}${path}/${id}`;
       },
 
       urlForDeleteRecord(id) {
-        let { path } = deletePath;
-        path = path.slice(1, path.indexOf('{') - 1);
+        let path = deletePath.path.slice(1, deletePath.path.indexOf('{') - 1);
```

Comment:
Nitty 'these can be `const` now' - if you need to come back in here maybe switch those out.